src/site/content/en/blog/building-a-settings-component/index.md

```diff
@@ -0,0 +1,959 @@
+
+/* take a slider element, return a percentage string for use in CSS */
+const rangeToPercent = slider => {
+  const max = slider.getAttribute('max') || 10;
+  const percent = slider.value / max * 100;
+
+  return `${parseInt(percent)}%`;
+};
+
+/* on page load, set the fill amount */
+sliders.forEach(slider => {
+  slider.style.setProperty('--track-fill', rangeToPercent(slider));
+
+  /* when a slider changes, update the fill prop */
+  slider.addEventListener('input', e => {
+    e.target.style.setProperty('--track-fill', rangeToPercent(e.target));
+  })
+})
+```
+
+I think this makes for a nice visual upgrade. The slider works great without
+Javascript, the `--track-fill` prop is not required, it simply will not have a
+fill style if not present. If Javascript is available, populate the custom
+property while also observing any user changes, syncing the custom property with
+the value. 
+
+[Here's a great
+post](https://css-tricks.com/sliding-nightmare-understanding-range-input/) on
+[CSS-Tricks](https://css-tricks.com/) by [Ana
+Tudor](https://twitter.com/anatudor), that demonstrates a CSS only solution for
+track fill. I also found this [Native UI
+range](https://app.native-elements.dev/editor/elements/range) very inspiring.
+
+### Thumb styles
+
+```css
+input[type="range"]::-webkit-slider-thumb {
+  appearance: none; /* clear styles, make way for mine */
+  cursor: ew-resize; /* cursor style to support drag direction */
+  border: 3px solid var(--surface3);
+  block-size: var(--thumb-size);
+  inline-size: var(--thumb-size);
+  margin-top: var(--thumb-offset);
+  border-radius: 50%;
+  background: var(--brand-bg-gradient) fixed;
+}
+```
+
+The majority of these styles are to make a nice circle. 
+Again you see the fixed background gradient there that unifies the
+dynamic colors of the thumbs, tracks and associated SVG elements.
+I separated the styles for the interaction to help isolate the `box-shadow`
+technique being used for the hover highlight:
+
+```css
+@custom-media --motionOK (prefers-reduced-motion: no-preference);
+
+::-webkit-slider-thumb {
+  …
+
+  /* shadow spread is initally 0 */
+  box-shadow: 0 0 0 var(--thumb-highlight-size) var(--thumb-highlight-color);
+  
+  /* if motion is OK, transition the box-shadow change */
+  @media (--motionOK) {
+    & {
+      transition: box-shadow .1s ease;
+    }
+  }
+
+  /* on hover/active state of parent, increase size prop */
+  @nest input[type="range"]:is(:hover,:active) & {
+    --thumb-highlight-size: 10px;
+  }
+}
+```
+
+{% Aside 'key-term' %} 
+[@custom-media](https://drafts.csswg.org/mediaqueries-5/#custom-mq) 
+is a Level 5 spec addition that
+[PostCSS Custom Media](https://github.com/postcss/postcss-custom-media), part of
+[PostCSS Preset Env](https://preset-env.cssdb.org/features). 
+{% endAside %}
+
+The goal was an easy to manage and animated visual highlight for user feedback.
+By using a box shadow I can prevent effecting layout with the effect. I do this
```

Comment:
```suggestion
The goal was an easy to manage and animated visual highlight for user feedback.
By using a box shadow I can avoid triggering layout with the effect. I do this
```

Reply:
might be good to add a link about what this means, maybe https://web.dev/animations-guide/ or there might be something you know of that is more focussed.